core/src/hir/attrs.rs

```diff
@@ -0,0 +1,157 @@
+                                    "`disable` not supported in backend {}",
+                                    validator.primary_name()
+                                )))
+                            } else {
+                                this.disable = true;
+                            }
+                        } else {
+                            errors.push(LoweringError::Other(format!(
+                                "Unknown diplomat attribute {p:?}: expected one of: `disable`"
+                            )));
+                        }
+                    }
+                    other => {
+                        errors.push(LoweringError::Other(format!(
+                            "Unknown diplomat attribute {other:?}: expected one of: `disable`"
+                        )));
+                    }
+                }
+            }
+        }
+
+        this
+    }
+}
+
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct BackendAttrSupport {
+    disabling: bool,
+    // more to be added: rename, namespace, etc
+}
+
+/// Defined by backends when validating attributes
+pub trait AttributeValidator {
+    /// The primary name of the backend, for use in diagnostics
+    fn primary_name(&self) -> &str;
+    /// Does this backend satisfy `cfg(backend_name)`?
+    /// (Backends are allowed to satisfy multiple backend names, useful when there
+    /// are multiple backends for a language)
+    fn is_backend(&self, backend_name: &str) -> bool;
+    /// does this backend satisfy cfg(name = value)?
+    fn is_name_value(&self, name: &str, value: &str) -> bool;
+    /// What backedn attrs does this support?
+    fn attrs_supported(&self) -> BackendAttrSupport;
+
+    /// Provided, checks if type satisfies a `DiplomatAttrCfg`
+    fn satisfies_cfg(&self, cfg: &DiplomatAttrCfg) -> bool {
+        match *cfg {
+            DiplomatAttrCfg::Not(ref c) => !self.satisfies_cfg(c),
+            DiplomatAttrCfg::Any(ref cs) => cs.iter().any(|c| self.satisfies_cfg(c)),
+            DiplomatAttrCfg::All(ref cs) => cs.iter().all(|c| self.satisfies_cfg(c)),
+            DiplomatAttrCfg::Star => true,
+            DiplomatAttrCfg::BackendName(ref n) => self.is_backend(n),
+            DiplomatAttrCfg::NameValue(ref n, ref v) => self.is_name_value(n, v),
+        }
+    }
+
+    // Provided, constructs an attribute
+    fn attr_from_ast(
+        &self,
+        ast: &ast::Attrs,
+        context: AttributeContext,
+        errors: &mut Vec<LoweringError>,
+    ) -> Attrs {
+        Attrs::from_ast(ast, self, context, errors)
+    }
+}
+
+/// A basic attribute validator
+#[non_exhaustive]
+#[derive(Default)]
+pub struct BasicAttributeValidator {
+    /// The primary name of this backend (should be unique, ideally)
+    pub backend_name: String,
+    /// The attributes supported
+    pub support: BackendAttrSupport,
+    /// Additional names for this backend
+    pub other_backend_names: Vec<String>,
```

Comment:
Thought: `other_backend_names` seems really niche but I guess it's ok. Maybe what we're looking for is a difference between the target language and the specific backend generating that language. For example, we currently have two C/C++ generators, and I imagine that people may want to implement "better" versions of generators for other target languages, too.